utils/findTopSellingProduct.js

```diff
@@ -0,0 +1,30 @@
+function findTopSellingProduct(salesData) {
+  // Use reduce to accumulate sales totals
+  const sales = salesData.reduce((acc, item) => {
+    const product = item.product;
+    const totalPrice = item.quantity * item.price;
+
+    if (acc[product]) {
+      acc[product] += totalPrice;
+    } else {
+      acc[product] = totalPrice;
+    }
+
+    return acc;
+  }, {});
+
+  // Use Object.entries and reduce to find the top product
```

Comment:
Reduce time complexity for this. You might not beed to use reduce twice

Reply:
please review, now it reduces time complexity by combining the sales calculation and top product determination into a single loop, eliminating the need for a second pass through the data.